src/manager/lotus.rs

```diff
@@ -1,24 +1,59 @@
 use std::collections::HashMap;
+use std::str::FromStr;
 
-use anyhow::Result;
+use crate::jsonrpc::JsonRpcClient;
+use crate::lotus::{LotusClient, LotusJsonRPCClient, MpoolPushMessage, NetworkVersion};
+use anyhow::{anyhow, Result};
 use async_trait::async_trait;
+use cid::Cid;
+use fil_actors_runtime::{builtin::singletons::INIT_ACTOR_ADDR, cbor};
 use fvm_shared::{address::Address, econ::TokenAmount};
 use ipc_gateway::Checkpoint;
 use ipc_sdk::subnet_id::SubnetID;
-use ipc_subnet_actor::{ConstructParams, JoinParams};
-
+use ipc_subnet_actor::{ConstructParams, JoinParams, types::MANIFEST_ID};
+use crate::manager::params::{ExecParams, INIT_EXEC_METHOD_NUM};
 use super::subnet::{SubnetInfo, SubnetManager};
 
-pub struct LotusSubnetManager {}
+pub struct LotusSubnetManager<T: JsonRpcClient> {
+    lotus_client: LotusJsonRPCClient<T>,
+}
 
 #[async_trait]
-impl SubnetManager for LotusSubnetManager {
-    async fn create_subnet(
-        &self,
-        _from: Address,
-        _params: ConstructParams,
-    ) -> Result<Address> {
-        panic!("not implemented")
+impl<T: JsonRpcClient + Send + Sync> SubnetManager for LotusSubnetManager<T> {
+    async fn create_subnet(&self, from: Address, params: ConstructParams) -> Result<Address> {
+        let network_name = self.lotus_client.state_network_name().await?;
+        if params.parent.to_string() != network_name {
+            return Err(anyhow!("parent network name not match"));
+        }
+
+        let actor_code_cid = self.get_subnet_actor_code_cid().await?;
+        let constructor_params = cbor::serialize(&params, "create subnet actor")?;
+
+        let exec_params = ExecParams {
+            code_cid: actor_code_cid,
+            constructor_params,
+        };
+        log::debug!("create subnet for init actor with params: {exec_params:?}");
+        let init_params = cbor::serialize(&exec_params, "init subnet actor params")?;
+        let message = MpoolPushMessage::new(
+            INIT_ACTOR_ADDR,
+            from,
+            INIT_EXEC_METHOD_NUM,
+            init_params.to_vec(),
+        );
+
+        let mem_push_response = self.lotus_client.mpool_push_message(message).await?;
+        let message_cid = mem_push_response.cid()?;
+        let nonce = mem_push_response.nonce;
+        log::debug!(
+            "create subnet message published with cid: {message_cid:?} and nonce: {nonce:?}"
+        );
+
+        let state_wait_response = self.lotus_client.state_wait_msg(message_cid, nonce).await?;
+        let address_raw = state_wait_response.receipt.r#return;
+        log::info!("created subnet at address: {address_raw:}");
+
+        Ok(Address::from_str(&address_raw)?)
```

Comment:
Have you tested this? Are we sure this type is returned in a string form? It would make sense as it is the type returned by the RPC call, but I would double-check it just in case.